doc/developer/filepoints.md

```diff
@@ -0,0 +1,64 @@
+# Introduction
+
+Failpoints are a way to force instrumented code to take a particular action if a given code line is reached while the failpoint is enabled. This is useful when writing tests -- for example, Materialize can be forced to panic or I/O calls can be made to return an error.
```

Comment:
You can cut "are a way to"

I'd make the second sentence "When writing tests, you can use failpoints to force Materialize to panic or to make I/O calls return an error."